Src/AutoFixture/MailAddressGenerator.cs

```diff
@@ -31,16 +31,47 @@ public class MailAddressGenerator : ISpecimenBuilder
         /// </remarks>
         public object Create(object request, ISpecimenContext context)
         {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
             if (!typeof(MailAddress).Equals(request))
             {
                 return new NoSpecimen(request);
             }
 
-            var name = Guid.NewGuid().ToString();
-            var host = fictitiousDomains[(uint)name.GetHashCode() % 3];
+            try
+            {
+                return TryCreateMailAddress(request, context);
+            }
+            catch (ArgumentNullException)
+            {
+                return new NoSpecimen(request);
+            }
+            catch (ArgumentException)

```

Comment:
It still works correctly, `ArgumentNullException` derives from `ArgumentException` so here we should only catch the latter.